[BUG] 出现空指针错误导致程序强行退出

### 请确认以下事项

- [x] 我已确认阅读并同意 [AGPL-3.0 第15条](https://www.gnu.org/licenses/agpl-3.0.txt#:~:text=15.%20Disclaimer%20of%20Warranty.) 。
本程序不提供任何明示或暗示的担保，使用风险由您自行承担。

- [x] 我已确认阅读并同意 [AGPL-3.0 第16条](https://www.gnu.org/licenses/agpl-3.0.txt#:~:text=16.%20Limitation%20of%20Liability.) 。
无论何种情况，版权持有人或其他分发者均不对使用本程序所造成的任何损失承担责任。

- [x] 我确认我的描述清晰，语法礼貌，能帮助开发者快速定位问题，并符合社区规则。

- [x] 我已确认阅读了[OpenList文档](https://doc.oplist.org)。

- [x] 我已确认没有重复的问题或讨论。

- [x] 我已确认是`OpenList`的问题，而不是其他原因（例如 [网络](https://doc.oplist.org/faq/howto#tls-handshake-timeout-read-connection-reset-by-peer-dns-lookup-failed-connect-connection-refused-client-timeout-exceeded-while-awaiting-headers-no-such-host-1) ，`依赖`或`操作`）。

- [x] 我认为此问题必须由`OpenList`处理，而非第三方。

- [x] 我已确认这个问题在最新版本中没有被修复。


### OpenList 版本（必填）

4.14

### 使用的存储驱动（必填）

189cloudpc

### 问题描述（必填）

登陆后，或点击天翼云盘驱动后，出现空指针报错，程序强制退出，前端显示Network Error,如图。

<img width="2122" height="1160" alt="Image" src="https://github.com/user-attachments/assets/d26ae68e-ca63-482d-82e5-cf103920a923" />

### 配置文件内容（必填）

默认配置

### 日志（可选）

myarch@myarch-jiguangxseriesgm6aq7c /u/bin [2]> sudo /usr/bin/openlist server --data /home/myarch/openlist
INFO[2025-10-05 13:20:39] reading config file: /home/myarch/openlist/config.json
INFO[2025-10-05 13:20:39] load config from env with prefix: OPENLIST_
INFO[2025-10-05 13:20:39] max buffer limit: 786MB
INFO[2025-10-05 13:20:39] mmap threshold: 4MB
INFO[2025-10-05 13:20:39] init logrus...
start HTTP server @ 0.0.0.0:52415
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x19b0ff1]

goroutine 39 [running]:
github.com/OpenListTeam/OpenList/v4/drivers/189pc.(*Cloud189PC).EncryptParams(0x6d84f20?, 0x0, 0x65?)
github.com/OpenListTeam/OpenList/v4/drivers/189pc/utils.go:82 +0x51
github.com/OpenListTeam/OpenList/v4/drivers/189pc.(*Cloud189PC).request(0xc000ba6508, {0x24be16e, 0x36}, {0x24158cd, 0x3}, 0xc00088add8, 0x0, {0x696b740, 0xc00054ea80}, {0x0, ...})
github.com/OpenListTeam/OpenList/v4/drivers/189pc/utils.go:96 +0x489
github.com/OpenListTeam/OpenList/v4/drivers/189pc.(*Cloud189PC).get(...)
github.com/OpenListTeam/OpenList/v4/drivers/189pc/utils.go:141
github.com/OpenListTeam/OpenList/v4/drivers/189pc.(*Cloud189PC).getCapacityInfo(0xc000ba6508, {0x6f55d50, 0xc00054e9a0})
github.com/OpenListTeam/OpenList/v4/drivers/189pc/utils.go:1479 +0xab
github.com/OpenListTeam/OpenList/v4/drivers/189pc.(*Cloud189PC).GetDetails(0xc000ba6508, {0x6f55d50?, 0xc00054e9a0?})
github.com/OpenListTeam/OpenList/v4/drivers/189pc/driver.go:415 +0x26
github.com/OpenListTeam/OpenList/v4/internal/op.GetStorageDetails({0x6f55d50?, 0xc00054e9a0?}, {0x6f63558?, 0xc000ba6508?})
github.com/OpenListTeam/OpenList/v4/internal/op/storage.go:447 +0x54
github.com/OpenListTeam/OpenList/v4/internal/op.GetStorageVirtualFilesWithDetailsByPath.func1({0x6f63558, 0xc000ba6508}, {0x6f61db8, 0xc000bbb400})
github.com/OpenListTeam/OpenList/v4/internal/op/storage.go:357 +0x15d
github.com/OpenListTeam/OpenList/v4/internal/op.getStorageVirtualFilesByPath.func2()
github.com/OpenListTeam/OpenList/v4/internal/op/storage.go:404 +0x94
created by github.com/OpenListTeam/OpenList/v4/internal/op.getStorageVirtualFilesByPath in goroutine 176
github.com/OpenListTeam/OpenList/v4/internal/op/storage.go:402 +0x6e5

### 复现链接（可选）

[这是data.db](https://github.com/user-attachments/files/22705947/data.txt) 下载后把文件后缀改回.db,放到相应目录里

--- a/drivers/189_tv/utils.go
+++ b/drivers/189_tv/utils.go
@@ -70,6 +70,9 @@ func (y *Cloud189TV) request(url, method string, callback base.ReqCallback, para
 }
 
 func (y *Cloud189TV) requestWithRetry(url, method string, callback base.ReqCallback, params map[string]string, resp interface{}, retryCount int, isFamily ...bool) ([]byte, error) {
+	if y.tokenInfo == nil {
+		return nil, fmt.Errorf("login failed")
+	}
 	req := y.client.R().SetQueryParams(clientSuffix())
 
 	if params != nil {

--- a/drivers/189pc/utils.go
+++ b/drivers/189pc/utils.go
@@ -90,6 +90,9 @@ func (y *Cloud189PC) EncryptParams(params Params, isFamily bool) string {
 }
 
 func (y *Cloud189PC) request(url, method string, callback base.ReqCallback, params Params, resp interface{}, isFamily ...bool) ([]byte, error) {
+	if y.getTokenInfo() == nil {
+		return nil, fmt.Errorf("login failed")
+	}
 	req := y.getClient().R().SetQueryParams(clientSuffix())
 
 	// 设置params

